packages/fast-components-styles-msft/src/utilities/high-contrast.ts

```diff
@@ -0,0 +1,219 @@
+import { focusOutlineWidth, outlineWidth } from "./design-system";
+
+enum HighContrastColor {
+    text = "WindowText",
+    hyperLinks = "LinkText",
+    disabledText = "GrayText",
+    selectedText = "HighlightText",
+    selectedBackground = "Highlight",
+    buttonText = "ButtonText",
+    buttonBackground = "ButtonFace",
+    background = "Background",
+}
+
+export const highContrastSelector: string = "@media (-ms-high-contrast:active)";
+
+export function applyHighContrastAdjustStealth(): CSSRules<DesignSystem> {
+    return {
+        [highContrastSelector]: {
+            background: HighContrastColor.buttonBackground,
+            color: HighContrastColor.buttonText,
+            fill: HighContrastColor.buttonText,
+            border: "none",
+            "-ms-high-contrast-adjust": "none",
+        },
+    };
+}
+
+export function applyHighContrastAdjustOutline(): CSSRules<DesignSystem> {
+    return {
+        [highContrastSelector]: {
+            background: HighContrastColor.buttonBackground,
+            borderColor: HighContrastColor.buttonText,
+            color: HighContrastColor.buttonText,
+            fill: HighContrastColor.buttonText,
+            "-ms-high-contrast-adjust": "none",
+        },
+    };
+}
+
+export function applyHighContrastBorder(): CSSRules<DesignSystem> {
+    return {
+        [highContrastSelector]: {
+            background: HighContrastColor.buttonBackground,
+            border: format("{0} solid ButtonText", toPx<DesignSystem>(outlineWidth)),
+        },
+    };
+}
+
+export function applyHighContrastAdjustSelectOutline(): CSSRules<DesignSystem> {
+    return {
+        [highContrastSelector]: {
+            background: HighContrastColor.selectedBackground,
+            borderColor: HighContrastColor.selectedBackground,
+            color: HighContrastColor.selectedText,
+            "-ms-high-contrast-adjust": "none",
+        },
+    };
+}
+
+export function applyHighContrastDisabled(): CSSRules<DesignSystem> {
+    return {
+        [highContrastSelector]: {
+            opacity: "1",
+            background: HighContrastColor.background,
+            color: HighContrastColor.disabledText,
+            fill: HighContrastColor.disabledText,
+        },
+    };
+}
+
+export function applyHighContrastDisabledBorder(): CSSRules<DesignSystem> {
+    return {
+        [highContrastSelector]: {
+            opacity: "1",
+            background: HighContrastColor.background,
+            borderColor: HighContrastColor.disabledText,
+            color: HighContrastColor.disabledText,
+            fill: HighContrastColor.disabledText,
+        },
+    };
+}
+
+export function applyHighContrastDisabledColor(): CSSRules<DesignSystem> {
+    return {
+        [highContrastSelector]: {
+            opacity: "1",
+            color: HighContrastColor.disabledText,
+        },
+    };
+}
+
+export function applyHighContrastDisabledFill(): CSSRules<DesignSystem> {
```

Comment:
Can fill and color go together and be "foreground"? See related notes below.

Reply:
update to applyHighContrastDisabledForeground